src/composer.ts

```diff
@@ -370,9 +379,10 @@ export class AtomicTransactionComposer {
     }
 
     const resolvedRefIndexes: number[] = [];
-    const foreignAccounts: string[] = [];
-    const foreignApps: number[] = [];
-    const foreignAssets: number[] = [];
+    const foreignAccounts: string[] = appAccounts == null ? [] : appAccounts;
+    const foreignApps: number[] = appForeignApps == null ? [] : appForeignApps;
```

Comment:
should this be `=== undefined`? I'm actually not sure

Reply:
I think that's fine too, `== null` is a looser check used later in this file so I followed the previous conventions here: https://github.com/algorand/js-algorand-sdk/pull/725/files#diff-534b9601ad1de4bb83d07dc0cb5af85424ccf602bcd8f4c0a2d3b79def06e8aeR458